app/common/telegram/MainButton.tsx

```diff
@@ -33,18 +33,16 @@ export const MainButton = ({ disabled, progress, hidden, color, textColor, text
     webApp?.MainButton.onClick(clickHandler);
 
     return () => {
-      hideButtonTimeout = setTimeout(() => webApp?.MainButton.hide(), 50);
+      hideButtonTimeout = setTimeout(() => webApp?.MainButton.hide(), 0);
```

Comment:
Why 0?

Reply:
Seems to me that moving back from page with `MainButton` to page without `MainButton` hides button much faster with value `0`. But if you've got strong argument agains `0`, let me know 😉